collectors/0/flume.py

```diff
@@ -92,13 +92,13 @@ def main(argv):
     err("This collector requires the `json' Python module.")
     return 1
 
-  def printmetric(metric, value, **tags):
+  def printmetric(component, metric, value, **tags):
     if tags:
       tags = " " + " ".join("%s=%s" % (name, value)
                             for name, value in tags.iteritems())
     else:
       tags = ""
-    print ("flume.%s %d %s %s" % (metric, ts, value, tags))
+    print ("flume.%s.%s %d %s %s" % (component, metric, ts, value, tags))
```

Comment:
Okay I figured it out - the 'metric' on line 109 is actually the component name e.g. "SINK.kafka-sink" and the 'key' is just the metric name suffix e.g. AppendReceivedCount. So to get the metrics the way they were we'd want the metric name in OpenTSDB to look like "flume.sink.AppendReceivedCount" - that requires the parameters to be passed the way they are in this PR. 

Happy to change some variable names if the code is unclear.